### ISSUE TITLE
Illegal LHS for expression definition: incompatibility between 4.1 and 4.2

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#293]
FORM 4.2 doesn't work with the following program
```
L F := 1;
P;
.end
```
which was valid in 4.1.

This change was introduced in 84d7188:
```
BAD  *   54f5a4d [2014-06-02] Make sure that there is no other stuff @vermaseren
     |\
GOOD | * 3d69423 [2014-05-28] Partially fix "make check" @Takahiro Ueda
GOOD | * 3ea180a [2014-05-26] Remove compilation warnings on gcc (Ubuntu 4.8.2-19ubuntu1) @Takahiro Ueda
BAD  * | de02800 [2014-06-02] Added Occurs() in f, quadfortran in format, isdefined in #if and toupper_ and tolower_ in macro definitions. @vermaseren
BAD  * | 84d7188 [2014-06-02] Occurs() in if statement. QuadFortran. toupper_ tolower_ in prepro. isdefined. @vermaseren
     |/
GOOD * ddad767 [2014-05-22] Fix "a strange problem" reported by Kostja @Takahiro Ueda
```

**Comment by tueda** [comment]
Is this a bug or intended behaviour? What is the meaning of the comment "The dummy accepted L F := RHS;"?

**Comment by vermaseren** [comment]
Version 4.2 is syntactically correct.
The proper language says L F = 1;
The fact that it did accept L F := 1; is obviously unintended behaviour.
Actually the manual mentions that a local statement consists of
the keyword local, a name of an expression an equal sign an expression
and a ;. It does not mention the possibility of using something different.

It happens occasionally that the syntax checker does not catch certain
unexpected constructs. It is not a good thing when people start using
those, rather than first asking about them, because that would make it very
difficult to plan future development of the language.

In short: sorry that it gives you maybe some extra work, but it cannot
be helped.

Jos

**Comment by tueda** [comment]
Thanks for the explanation, Jos. Yes, the behaviour of version 4.2 looks saner. I'm closing this issue.
